bundles/org.openhab.binding.lifx/src/main/java/org/openhab/binding/lifx/internal/handler/LifxLightHandler.java

```diff
@@ -486,121 +513,129 @@ private void sendPacket(Packet packet) {
+            } else {
+                compositeCommandsRunnable.run();
             }
         }
     }
 
+    private @Nullable Runnable getChannelCommandRunnable(ChannelUID channelUID, Command command) {
+        switch (channelUID.getId()) {
+            case CHANNEL_BRIGHTNESS:
+                if (command instanceof PercentType) {
+                    return () -> handlePercentCommand((PercentType) command);
+                } else if (command instanceof OnOffType) {
+                    return () -> handleOnOffCommand((OnOffType) command);
+                } else if (command instanceof IncreaseDecreaseType) {
+                    return () -> handleIncreaseDecreaseCommand((IncreaseDecreaseType) command);
+                } else {
+                    return null;
+                }
+            case CHANNEL_COLOR:
+                if (command instanceof HSBType) {
+                    return () -> handleHSBCommand((HSBType) command);
+                } else if (command instanceof PercentType) {
+                    return () -> handlePercentCommand((PercentType) command);
+                } else if (command instanceof OnOffType) {
+                    return () -> handleOnOffCommand((OnOffType) command);
+                } else if (command instanceof IncreaseDecreaseType) {
+                    return () -> handleIncreaseDecreaseCommand((IncreaseDecreaseType) command);
+                } else {
+                    return null;
+                }
+            case CHANNEL_EFFECT:
+                if (command instanceof StringType && features.hasFeature(TILE_EFFECT)) {
+                    return () -> handleTileEffectCommand((StringType) command);
+                } else {
+                    return null;
+                }
+            case CHANNEL_HEV_CYCLE:
+                if (command instanceof OnOffType) {
+                    return () -> handleHevCycleCommand((OnOffType) command);
+                } else {
+                    return null;
+                }
+            case CHANNEL_INFRARED:
+                if (command instanceof PercentType) {
+                    return () -> handleInfraredCommand((PercentType) command);
+                } else if (command instanceof IncreaseDecreaseType) {
+                    return () -> handleIncreaseDecreaseInfraredCommand((IncreaseDecreaseType) command);
+                } else {
+                    return null;
+                }
+            case CHANNEL_TEMPERATURE:
+                if (command instanceof PercentType) {
+                    return () -> handleTemperatureCommand((PercentType) command);
+                } else if (command instanceof IncreaseDecreaseType) {
+                    return () -> handleIncreaseDecreaseTemperatureCommand((IncreaseDecreaseType) command);
+                } else {
+                    return null;
+                }
+            default:
+                try {
+                    if (channelUID.getId().startsWith(CHANNEL_COLOR_ZONE)) {
```

Comment:
It make sense to split up this part or other parts of this method, given it became pretty big.

Reply:
Yes this whole class is actually getting too big so it means it is doing too much. Creating some subclasses for each thing type might result in nicer code and it would also group related code. :slightly_smiling_face: